README.md

```diff
@@ -193,3 +193,61 @@ $ telnet 192.168.99.100 1234
 
 This `-p` argument links port 4321 on the container to port 1234 on the VM network adapter. It must appear after `docker run` and before `dmscid/plankton`. This allows us to connect to the container from outside of the VM, in this case using Telnet. The `192.168.99.100` IP is the IP of the VM on the bridge network between the host and the VM. VirtualBox will typically use this IP when available, but it may be different on your system.
 
+## Remote access to devices
+
+*Please note that this functionality should only be used on a trusted network.*
+
+Besides the device specific protocols, the device can be made accessible from the outside via JSON-RPC over ZMQ. This can be achieved by passing the `-r` option with a `host:port` string to the simulation:
+
+```
+$ python simulation.py -r 127.0.0.1:10000 -d chopper -- -p SIM:
+```
+
+Now the device can be controlled via the `control.py`-script in a different terminal window. The service can be queried to show the available objects using the `-l` or `--list-objects` flag:
+
+```
+$ python control.py --ip 127.0.0.1 --port 10000 --list-objects
+```
+
+The `ip` and `port` parameters default to the values shown here, so they will be omitted in the following examples. To get information on the API of an object, the `-a` or `--show-api` option can be used in conjunction with a device name obtained from the previous command:
+
+```
+$ python control.py -a device
+```
+
+This will output a list of properties and methods which is available for remote access. This may not comprise the full interface of the object depending on the server side configuration. Obtaining the value of a property is done like this:
+
+```
+$ python control.py device state
+```
+
+The same syntax is used to call methods without parameters:
+
+```
+$ python control.py device initialize
+```
+
+To set a property to a new value, the value has to be supplied on the command line:
+
+```
+$ python control.py device targetSpeed 100
+$ python control.py device start
+```
+
+Only numeric types and strings can be used as arguments via the `control.py`-script. The script always tries to convert parameters to `int` first, then to `float` and leaves it as `str` if both fail. For other types and more control over types, it's advised to write a Python script instead using the tools provided in `core.rpc_client` which makes it possible to use the remote objects more or less transparently. An example to control the chopper:
+
+```
+from time import sleep
+from core.rpc_client import ZMQJSONRPCConnection, get_remote_object_collection
```

Comment:
I agree on the naming. `ExposedObject` was named strangely before as well (and stuff was differently organized as well). In fact I already renamed everything according to your suggestions, I like those names. They fit better 

I also agree that those methods should be moved into `ControlClient`. The problem with the `dict`-like interface is that the server does not necessarily expose a collection of named objects, it can also expose a single object, which is then unnamed. In that case (and I think it's a valid use case) it would not make sense to access it like a `dict`, this is basically the only reason why those two methods exist separately. What I do not like about this is that you have to know it beforehand, at least partly. Since `ExposedObjectCollection` inherits from `ExposedObject`, you can always go that route (using the current scheme):

```python
obj = get_remote_object(connection)
object_names = obj.get_objects() # May fail if obj is not in fact a collection

objects = [get_remote_object(connection, name) for name in object_names]
```

So the client needs to do some sort of check what the server actually exposes, so it might not be as easy as a `dict`-like interface.

My suggestion would be to move `get_remote_object` into `ControlClient` and keep a small function that creates a dict from that object, like `get_remote_object_collection` does now:

```python
connection = ControlClient(host='...', port='...')
objects = remote_collection(connection.get_object())
```

The other alternative I see would be that the `dict` like interface is just empty if the remote object is not of the right type, but at the moment that does not convince me, because you still need a way to get the top level object, so there are two interfaces in one...